src/Services/Settings.js

```diff
@@ -6,6 +6,7 @@ const settings = require('electron-settings');
 const oldDefaultMnemonic =  "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
 
 const initialSettings = {
+  validationErrors: {}, // of the format {SETTING_NAME (i.e. hosname would be server.hostname): "error text"}
```

Comment:
Comment here is confusing. I don't know whether it should be `validationErrors: { server: { hostname: "error text" } }` or `validationErrors: { "server.hostname": "error text" }`.

Also isn't this state, not settings? Anything in the `initialSettings` blob should be expected to persist to the settings file. I don't think we want settings which define whether other settings are erroneous, do we? We have application logic for that.

Reply:
I'm going to address this by removing `validationErrors` from `initialSettings` but change the `state` from being `state.settings` to `state.config` which has a `settings` object as well as a `validationErrors` object. This keeps `validationErrors` as part of the state while not mixing it in with the `SettingsService`